resources/assets/js/components/H2Components/Dialog.tsx

```diff
@@ -20,101 +22,97 @@ const DialogContext = React.createContext<DialogContext | undefined>(undefined);
   return (
-    <div data-h2-dialog-actions className={className} {...rest}>
+    <div data-h2-dialog-actions {...rest}>
       {children}
     </div>
   );
 };
 
-const ActionBtn: React.FunctionComponent<GeneralBtnProps & GeneralProps> = (
-  props,
-) => {
+type ActionBtnProps = GeneralBtnProps;
+const ActionBtn: React.FunctionComponent<ActionBtnProps> = ({
+  buttonStyling,
+  children,
+  type,
+  ...rest
+}) => {
   const { id } = useDialogContext();
-  const { buttonStyling, type, onClick, className, children, ...rest } = props;
-  const ref = React.useRef(null);
-  React.useEffect((): void => {
-    h2DialogTrigger(ref.current);
-  });
   return (
     <button
       data-h2-dialog-trigger={`${id}`}
       data-h2-button={buttonStyling}
-      ref={ref}
       type={type || "button"}
-      onClick={onClick}
-      className={className}
       {...rest}
     >
       {children}
     </button>
   );
 };
 
-const Content: React.FunctionComponent<GeneralProps> = (props) => {
+type ContentProps = Div;
+const Content: React.FunctionComponent<ContentProps> = ({
+  children,
+  ...rest
+}) => {
   const { id } = useDialogContext();
-  const { className, children, ...rest } = props;
   return (
-    <div
-      data-h2-dialog-content
-      id={`${id}Content`}
-      className={className}
-      {...rest}
-    >
+    <div data-h2-dialog-content id={`${id}Content`} {...rest}>
       {children}
     </div>
   );
 };
 
-const Header: React.FunctionComponent<GeneralProps> = (props) => {
+type HeaderProps = Div;
+const Header: React.FunctionComponent<HeaderProps> = ({
+  children,
+  ...rest
+}) => {
   useDialogContext(); // Ensures sub-component can only be used within the Dialog component.
-  const { className, children, ...rest } = props;
   return (
-    <div data-h2-dialog-title className={className} {...rest}>
+    <div data-h2-dialog-title {...rest}>
       {children}
     </div>
   );
 };
 
-const Title: React.FunctionComponent<GeneralProps> = (props) => {
+type TitleProps = Heading;
+const Title: React.FunctionComponent<TitleProps> = ({ children, ...rest }) => {
   const { id } = useDialogContext();
-  const { className, children, ...rest } = props;
   return (
-    <h5 data-h2-focus id={`${id}Title`} className={className} {...rest}>
+    <h5 data-h2-focus id={`${id}Title`} {...rest}>
       {children}
     </h5>
   );
 };
 
-interface TriggerProps extends GeneralProps, GeneralBtnProps {
-  id: string;
-}
-
 /** This Trigger component opens the dialog and sits outside the main dialog component */
-const Trigger: React.FunctionComponent<TriggerProps> = (props) => {
-  const { id, buttonStyling, className, children, ...rest } = props;
-  const ref = React.useRef(null);
-  React.useEffect((): void => {
-    h2DialogTrigger(ref.current);
-  });
+type TriggerProps = GeneralBtnProps;
+const Trigger: React.FunctionComponent<TriggerProps> = ({
+  id,
+  buttonStyling,
+  children,
+  ...rest
+}) => {
   return (
     <button
+      aria-haspopup
```

Comment:
Looks to me like aria-haspopup needs [needs a value](https://www.digitala11y.com/aria-haspopup-properties/) of "true" or "dialog".